activerecord/lib/active_record/core.rb

```diff
@@ -548,7 +548,7 @@ def pretty_print(pp)
       return super if custom_inspect_method_defined?
       pp.object_address_group(self) do
         if defined?(@attributes) && @attributes
-          attr_names = self.class.attribute_names.select { |name| has_attribute?(name) }
+          attr_names = self.attribute_names.select { |name| has_attribute?(name) }
```

Comment:
(If following previous suggestion.)

```suggestion
          attr_names = attribute_names.select { |name| has_attribute?(name) }
```